### PR TITLE
fix: get and set subscriptions

### PR DESCRIPTION
**Pull request by saravanansivanathan**
## Description of this change

> Description here
Get and Set subscriptions from lanyard
## Why is this change being made?
- [ ] Chore (non-functional changes)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [ ] I have informed stakeholders of my changes.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1049,6 +1049,11 @@ export interface Configuration {
    */
   purposes?: Purpose[]
 
+  /**
+   * subscription configuration
+   */
+  subscriptionConfiguration?: GetSubscriptionConfigurationResponse
```
Comment (saravanansivanathan) [inline src/index.ts:1055]: `subscriptionConfiguration` was made available in configuration for lanyard to render the UI

File: src/index.ts [hunk 2]
```diff
@@ -1767,6 +1777,13 @@ export interface Ketch {
    */
   registerStorageProvider(policy: StorageOriginPolicy, provider: StorageProvider): Promise<void>
 
+  /**
+   * Sets subscriptions
+   *
+   * @param subscriptions SetSubscriptionsRequest
+   */
+  setSubscriptionsRequest(subscriptions: SetSubscriptionsRequest): Promise<void>
```
Comment (sethyates) [inline src/index.ts:1785]: `setSubscriptions`

Also need to add a `getSubscriptions` taking a `GetSubscriptionsRequest` and returning `GetSubscriptionsResponse`.

File: src/index.ts [hunk 3]
```diff
@@ -1049,6 +1049,16 @@ export interface Configuration {
    */
   purposes?: Purpose[]
 
+  /**
+   * User subscriptions
+   */
+  dataSubjectSubscriptions?: GetSubscriptionsResponse
```
Comment (sethyates) [inline src/index.ts:1055]: This does NOT belong on the configuration. This should be from `getSubscriptions` below.

**Comment by sethyates** [comment]
Might as well make this change too:
```
/**
 * ContactMethod
 */
export interface ContactMethod {
  name: string
}

/**
 * GetSubscriptionConfigurationResponse
 */
export interface GetSubscriptionConfigurationResponse {
  language: string
  organization: Organization
  property: Property
  identities: { [key: string]: Identity }
  contactMethods: { [key: string]: ContactMethod }
  controls: SubscriptionControl[]
  topics: SubscriptionTopic[]
}

```

**Pull request merged**